server.mjs

```diff
@@ -38,47 +42,48 @@ app.get("/forecast", async (_, res) => {
   }
 });
 
-// TODO: .get('/cityForecast/:cityName', (...) => {...})
-app.get("/cityForecast", async (req, res) => {
+// TODO: .get('/cityForecast/:cityName', (...) => {...})     (query parameter > route parameter)
+app.get("/cityForecast/:cityName", async (req, res) => {
   try {
-    const parsedUrl = await getParsedUrl(req);
-    const cityName = parsedUrl.searchParams.get("city");
+    const cityName = req.params.cityName.toLowerCase();
     if (cityName) {
       const cityForecast = await getForecastForCityIes(cityName);
-      res.writeHead(200, { "Content-Type": "application/json" });
-      res.end(cityForecast);
+      res.status(200).json(cityForecast);
     } else {
-      res.writeHead(400, { "Content-Type": "application/json" });
-      res.end(JSON.stringify({ error: "No city query parameter provided" }));
+      res.status(400).json({ error: "No city query parameter provided" });
     }
   } catch (error) {
+    console.log(error);
     res.status(500).send("Internal Server Error");
   }
 });
 
+app.get("/addCity", (req, res) => {
+  res.sendFile(path.join(__dirname, "templates/add_city.html"));
+});
 // TODO: move to post request (data would comes from body)
-app.get("/addCity", async (req, res) => {
+app.post("/postAddCity", async (req, res) => {
   try {
-    const parsedUrl = await getParsedUrl(req);
-    const latitude = parsedUrl.searchParams.get("latitude");
-    const longitude = parsedUrl.searchParams.get("longitude");
-    const city = parsedUrl.searchParams.get("city");
-    if (isNaN(+latitude) || isNaN(+longitude)) {
-      res.writeHead(200, { "Content-Type": "text/plain" });
-      res.end("Latitude and longitude must be numbers");
-    }
-    if (city && latitude && longitude) {
+    const city = req.body.city;
+    const latitude = req.body.latitude;
+    const longitude = req.body.longitude;
+    if (isNaN(+latitude) || isNaN(+longitude) || !isNaN(city)) {
+      res
+        .status(400)
+        .send("City must be string ,latitude and longitude must be numbers");
+    } else if (city && latitude && longitude) {
       await addCity(city, latitude, longitude);
-      res.writeHead(200, { "Content-Type": "text/plain" });
-      res.end("City added successfully");
+      res.status(200).send("City added successfully");
+    } else {
+      res.status(400).send("Please provide city, latitude, and longitude");
     }
```

Comment:
Same story about conditions, we can use `if` and `return` right inside to not overcomplicate the logic
And we can omit last `else`. Our response message might just contain information, that provided data are incorrect
If you would like to add more context to it, you can just mention them right in the message, like
```
.send(`Invali data input. Allowed data: city [string] - ${city}, latitude [number] - ${latitude}, longitude [number] - ${longitude}`);
```
But maybe I would also recommend to check `yup` package. This way you wouldn't need to manage error string by yourself
![image](https://github.com/IhorYu/JsForecastBun/assets/33233554/a00d436f-c759-4600-b723-96c30c930297)

Reply:
cool, thank you for that, I'll take a look and will proceed with some updated in the next PR accordingly :)